spec/support/env.cr

```diff
@@ -1,12 +1,10 @@
 def with_env(values : Hash)
-  old_values = {} of String => String?
+  old_values = ENV.to_h
   begin
     values.each do |key, value|
       key = key.to_s
-      old_values[key] = ENV[key]?
       ENV[key] = value
     end
-
     yield
   ensure
     old_values.each do |key, old_value|
```

Comment:
I think we should use `replace` to reset the original state.

Reply:
@straight-shoota  
I had considered that, but since I was adding `.replace` with this PR, it felt self-referential

Reply:
That's true, but it's also necessary for proper behaviour. And the functionality of `replace` is tested explicitly, so this should be fine.

Reply:
I simplified `with_env` to use both `replace` and `merge!`.  One side effect of using `merge!` inside of `with_env` is that it exposed the cases in existing test-cases where:
+ the keys were symbols (because it came from keyword arguments) which `merge!` did not handle previously.
+ nil values were being merged using `with_env`, and `merge!` also did not handle this previously.

Code has been updated accordingly.  If you feel that `merge!` should not support symbolic keys or nil values in the *other* Hash, then let me know and I can back out those changes.